autonomi/src/client/mod.rs

```diff
@@ -52,33 +60,70 @@ const CLIENT_EVENT_CHANNEL_SIZE: usize = 100;
 // Amount of peers to confirm into our routing table before we consider the client ready.
 pub use ant_protocol::CLOSE_GROUP_SIZE;
 
+/// Events emitted by the client.
+#[derive(Debug, Clone)]
+pub enum ClientEvent {
+    /// A new peer was discovered.
+    PeerDiscovered(libp2p::PeerId),
+    /// A peer was disconnected.
+    PeerDisconnected(libp2p::PeerId),
+    /// Upload operation completed.
+    UploadComplete(UploadSummary),
+}
+
+/// Summary of an upload operation.
+#[derive(Debug, Clone)]
+pub struct UploadSummary {
+    pub record_count: usize,
+    pub tokens_spent: Amount,
+}
+
+/// Error returned by [`Client::init`].
+#[derive(Debug, thiserror::Error)]
+pub enum ConnectError {
+    /// Did not manage to populate the routing table with enough peers.
+    #[error("Failed to populate our routing table with enough peers in time")]
+    TimedOut,
+
+    /// Same as [`ConnectError::TimedOut`] but with a list of incompatible protocols.
+    #[error("Failed to populate our routing table due to incompatible protocol: {0:?}")]
+    TimedOutWithIncompatibleProtocol(HashSet<String>, String),
+
+    /// An error occurred while bootstrapping the client.
+    #[error("Failed to bootstrap the client")]
+    Bootstrap(#[from] ant_bootstrap::Error),
+}
+
+/// Client mode indicating read-only or read-write capabilities
+pub enum ClientMode {
+    /// Read-only mode without a wallet
+    ReadOnly,
+    /// Read-write mode with an attached wallet
+    ReadWrite(EvmWallet),
+}
+
+impl Clone for ClientMode {
+    fn clone(&self) -> Self {
+        match self {
+            Self::ReadOnly => Self::ReadOnly,
+            Self::ReadWrite(wallet) => Self::ReadWrite(wallet.clone()),
+        }
+    }
+}
+
 /// Represents a client for the Autonomi network.
-///
-/// # Example
-///
-/// To start interacting with the network, use [`Client::init`].
-///
-/// ```no_run
-/// # use autonomi::client::Client;
-/// # #[tokio::main]
-/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
-/// let client = Client::init().await?;
-/// # Ok(())
-/// # }
-/// ```
 #[derive(Clone)]
 pub struct Client {
     pub(crate) network: Network,
     pub(crate) client_event_sender: Arc<Option<mpsc::Sender<ClientEvent>>>,
     pub(crate) evm_network: EvmNetwork,
+    pub(crate) mode: ClientMode,
```

Comment:
What's the reason behind mode here?

Reply:
The mode is for local network mode (mens) or not.